src/Frame.js

```diff
@@ -0,0 +1,103 @@
+class Frame {
+  constructor(rolls = [], number = null) {
+    this.rolls = rolls
+    this.bonusRolls = []
+    this.number = number
+  }
+
+  getRoll(n) {
+    if (this.rolls[n] == null) {
+      return 0
+    } else {
+      return this.rolls[n]
+    }
+  }
+
+  getBonusRoll(n) {
+    if (this.bonusRolls[n] == null) {
+      return 0
+    } else {
+      return this.bonusRolls[n]
+    }
+  }
+
+  getNumber() {
+    return this.number
+  }
+
+  setNumber(n) {
+    this.number = n
+  }
+
+  roll(n) {
+    if (this.rolls.length < 2 && !this.isStrike()) {
+      this.rolls.push(n)
+    } else {
+      this.bonusRolls.push(n)
+    }
+  }
+
+  pins() {
+    return this.rolls.reduce((a, b) => a + b, 0)
+  }
+
+  isStrike() {
+    return this.rolls[0] == 10
+  }
+
+  isSpare() {
+    return this.pins() == 10 && !this.isStrike()
+  }
+
+  isComplete() {
+    if (this.number == 10 && this.pins() == 10) {
+      return this.rolls.length + this.bonusRolls.length == 3
+    } else {
+      return this.rolls.length == 2 || this.isStrike()
+    }
+  }
+
+  score(nextFrame1 = null, nextFrame2 = null) {
+    return this.pins() + this.bonus(nextFrame1, nextFrame2)
+  }
+
+  bonus(nextFrame1 = null, nextFrame2 = null) {
+    if (this.isStrike()) {
+      return this.nextRoll1(nextFrame1) + this.nextRoll2(nextFrame1, nextFrame2)
```

Comment:
I'm slightly obsessed with how well you've used the frames class and given it all the frame responsibilities, I honestly feel like you extracted this class so perfectly.